{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0f6193c4",
   "metadata": {},
   "source": [
    "## Libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "22a2ddad",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\enhi\\AppData\\Local\\Temp\\ipykernel_9092\\4077202506.py:4: UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:\n",
      "\n",
      "import os\n",
      "os.environ['USE_PYGEOS'] = '0'\n",
      "import geopandas\n",
      "\n",
      "In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).\n",
      "  import geopandas as gpd\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import rioxarray as rxr\n",
    "from shapely.geometry import mapping\n",
    "import geopandas as gpd\n",
    "import rasterio as rio\n",
    "from rioxarray.merge import merge_arrays"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4110198c",
   "metadata": {},
   "source": [
    "## The aim of this Notebook is to mosaic 4 tiles and clip them to AOI using long time series data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6aa3f183",
   "metadata": {},
   "outputs": [],
   "source": [
    "main_dir = r'C:\\Users\\enhi\\OneDrive - gmv.com\\Documents\\Projects\\ANIN\\Generating Indices\\FAPAR anomaly'\n",
    "# Output directory\n",
    "Clipped_path = os.path.join(main_dir, 'Clipped_FAPAR')\n",
    "# Directories for each tile\n",
    "Boundary = os.path.join(main_dir, 'Boundary')\n",
    "Tile_1 = os.path.join(main_dir, 'tile_1')\n",
    "Tile_2 = os.path.join(main_dir, 'tile_2')\n",
    "Tile_3 = os.path.join(main_dir, 'tile_3')\n",
    "Tile_4 = os.path.join(main_dir, 'tile_4')\n",
    "# Lists of images' names for each tile\n",
    "Images_in_tile1 = sorted(os.listdir(Tile_1))\n",
    "Images_in_tile2 = sorted(os.listdir(Tile_2))\n",
    "Images_in_tile3 = sorted(os.listdir(Tile_3))\n",
    "Images_in_tile4 = sorted(os.listdir(Tile_4))\n",
    "# Directory to an example of Copernicus FAPAR to take the shape and crs fro it\n",
    "Copernicus_ex =rxr.open_rasterio(r'C:\\Users\\enhi\\OneDrive - gmv.com\\Documents\\Projects\\ANIN\\Generating Indices\\FAPAR anomaly\\Monthly_FAPAR_Glass\\April\\2000April.tif', masked =True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "17913336",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shapefile loaded. To prepare for masking, run the function\n",
      "        `select_shape`.\n",
      "Mask created.\n"
     ]
    }
   ],
   "source": [
    "# Load the shapefile\n",
    "def load_shape_file(filepath):\n",
    "    \"\"\"Loads the shape file desired to mask a grid.\n",
    "    Args:\n",
    "        filepath: Path to *.shp file\n",
    "    \"\"\"\n",
    "    shpfile = gpd.read_file(filepath)\n",
    "    print(\"\"\"Shapefile loaded. To prepare for masking, run the function\n",
    "        `select_shape`.\"\"\")\n",
    "    return shpfile\n",
    "\n",
    "#Create the mask\n",
    "def select_shape(shpfile):\n",
    "\n",
    "    \"\"\"Select the submask of interest from the shapefile.\n",
    "    Args:\n",
    "        shpfile: (*.shp) loaded through `load_shape_file`\n",
    "        category: (str) header of shape file from which to filter shape.\n",
    "            (Run print(shpfile) to see options)\n",
    "        name: (str) name of shape relative to category.\n",
    "           Returns:\n",
    "        shapely polygon\n",
    "    \"\"\"\n",
    "\n",
    "    col_code = 'ISO3_CODE'\n",
    "    country_codes = ['ZAF', 'LSO', 'SWZ']\n",
    "\n",
    "    # Extract the rows that have 'ZAF', 'LSO', or 'SWZ' in the 'SOV_A3' column\n",
    "    selected_rows = shpfile[shpfile[col_code].isin(country_codes)]\n",
    "\n",
    "    # Combine the selected polygons into a single polygon\n",
    "    unioned_polygon = selected_rows.geometry.unary_union\n",
    "\n",
    "    # Convert the unioned polygon to a geopandas dataframe with a single row\n",
    "    mask_polygon = gpd.GeoDataFrame(geometry=[unioned_polygon])\n",
    "    \n",
    "    print(\"\"\"Mask created.\"\"\")\n",
    "\n",
    "    return mask_polygon\n",
    "shpfile = load_shape_file(Boundary + '\\\\'+ 'CNTR_RG_01M_2020_4326.shp')\n",
    "merged = select_shape(shpfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c54669c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# crs of FAPAR copernicus\n",
    "Copernicus_ex_crs = Copernicus_ex.rio.crs\n",
    "# Define a crs to the AOI \n",
    "merged.crs =Copernicus_ex_crs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "02c04a15",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n",
      "C:\\Users\\enhi\\Anaconda3\\envs\\rasterio\\Lib\\site-packages\\rioxarray\\raster_writer.py:132: UserWarning: The nodata value (3.402823466e+38) has been automatically changed to (3.4028234663852886e+38) to match the dtype of the data.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "# looping based on the number of images\n",
    "for i in range (0 , len(Images_in_tile4)):\n",
    "    # Open corresponding images from each tile\n",
    "    tile1 = rxr.open_rasterio(Tile_1 + '\\\\'+ Images_in_tile1[i], masked=True)\n",
    "    tile2 = rxr.open_rasterio(Tile_2 + '\\\\'+ Images_in_tile2[i], masked=True)\n",
    "    tile3 = rxr.open_rasterio(Tile_3 + '\\\\'+ Images_in_tile3[i], masked=True)\n",
    "    tile4 = rxr.open_rasterio(Tile_4 + '\\\\'+ Images_in_tile4[i], masked=True)\n",
    "    # Mosiac of the corresponding images of each tile by using only NDVI bands\n",
    "    merged_raster = merge_arrays(dataarrays = [tile1, tile2,\n",
    "                                               tile3,tile4])\n",
    "    # Clipping the mosiac to the AOI\n",
    "    clipped = merged_raster.rio.clip(merged.geometry.apply(mapping),\n",
    "                                     crs=merged.crs,\n",
    "                                     all_touched=True,\n",
    "                                     from_disk=True).squeeze()\n",
    "    # New directory includes output clipped images of each year\n",
    "    # Note that [9:13] is used as to extract the year from the name of images\n",
    "    year = Clipped_path + '\\\\'+ Images_in_tile1[i][16:20]\n",
    "    check_folder_year = os.path.isdir(year)\n",
    "    if not check_folder_year:\n",
    "        os.makedirs(year)\n",
    "    # Export clipped images\n",
    "    clipped.rio.to_raster(year + '\\\\'+ Images_in_tile1[i]+ '.tif')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "rasterio",
   "language": "python",
   "name": "rasterio"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
